### ISSUE TITLE
Hi, the calculation of performance in test.py is incorrect.

### ISSUE THREAD

**Issue by LMissher** [songyangco/STNN#1]
For instance, the correct code in test.py at line 60 is "out_denormalized_3 = out.detach().cpu().numpy()**[:, 3]**.flatten()".

I hope you can update codes to rerun experiments and update the new results in your paper.

**Comment by songyangco** [comment]
Thanks for raising your concern. However, I think the code is correct.

In our settings, we always predict the traffic conditions for the next 12 timesteps given the previous 12 timesteps as the input. For example, given [t1,...t12], we always predict the traffic conditions of [t13,...,t24]. The out_denormalized_3 means measuring the prediction performance of the immediate next 3 timesteps. In other words, t13, t14, t15. So the original code 
`out_denormalized_3 = out.detach().cpu().numpy()[:, 0:3].flatten()`
takes the first three timesteps (t13, t14, t15) out to evaluate. You are suggesting to use **[:, 3]** which means measuring the prediction performance of the fourth timesteps, and it is not our intention.

**Comment by LMissher** [comment]
I apologize for the **[:3]**, which should be **[:2]** to denote the results at t15, _i.e._, the results of short-term (15 mins). The meaning of short-term (15 mins), mid-term (30 mins), and long-term (60 mins) in all prior works are results at t15, t18, and t24 rather than the result between t1* and t13, _i.e._, should be **[x]** rather than **[:x]**.

GMAN:https://github.com/zhengchuanpan/GMAN/blob/master/METR/test.py

Graph WaveNet:https://github.com/nnzhan/Graph-WaveNet/blob/master/test.py

**Comment by songyangco** [comment]
Thanks for pointing out that. We use the merged metrics in our experiments which aligned to STGCN. For your question, I've added a pred_mode argument so you can choose the ideal mode you like.

**Comment by LMissher** [comment]
I think you should utilize the results of 'sep' mode in your paper rather than 'merge' mode. STGCN is not a baseline based on the METR-LA and PEMS-BAY datasets. Besides, DCRNN, Graph WaveNet, GMAN, and MTGCN use 'sep' mode, and they are baselines based on the METR-LA and PEMS-BAY datasets. It is not a fair comparison between them and your model. I hope you can update the results in your paper to obey academic ethics.
